### ISSUE TITLE
[Question]: Regarding Frequent Updates

### ISSUE THREAD

**Issue by Heet-Bhalodiya** [ueberdosis/tiptap#5379]
### Affected Packages

core, extension-color, extension-list-item, extension-placeholder, extension-text-align, extension-text-style, extension-underline, pm, react, starter-kit

### Version(s)

2.5.4

### Bug Description

We are experiencing issues with the frequent updates of the `Tiptap` editor, which affect our template and our customers' installations using package managers such as `npm`, `yarn`, or `pnpm`.

For instance, when we use Tiptap version `2.5.4` and version `2.5.5` is released, it results in installation warnings due to mismatched peer dependencies. The error indicates that packages require version 2.5.5, while we have version 2.5.4. Please see the attached screenshot for reference.

![image](https://github.com/user-attachments/assets/2c33d5b6-f303-4541-8498-e9a7f5a9ca7b)

This situation creates significant inconvenience for us and our customers, as even minor updates trigger these installation warnings.

Please note that we can't provide `^` in the `package.json` file in our project as it might cause any issue or break the application for customers.

### Browser Used

Chrome

### Expected Behavior

We kindly request that you address this issue to ensure smoother transitions with updates, preventing such warnings from occurring with each minor release.

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by nperez0111** [comment]
Why do you not just update all of the packages to 2.5.5? You have some of them as 2.5.4 they are meant to be installed together & that is why they claim peer dep on each other?

In your screenshot you have @tiptap/react@2.5.4 but @tiptap/core@2.5.5

I don't think that it is our problem that you can't resolve your dependencies, they are warnings you can even force your package manager to install them if you wanted to

**Comment by Heet-Bhalodiya** [comment]
Hello @nperez0111,

I have not updated my packages and am currently using version `2.5.4` The issue occurs when I specify version `2.5.4` in my `package.json` file, delete the lock file, and then reinstall. This process results in the warning shown in the above screenshot.

**Comment by nperez0111** [comment]
@Heet-Bhalodiya 
The way that we specify our deps is like this:
https://github.com/ueberdosis/tiptap/blob/22cb0389973fb52ca279634915374fd46ddf7724/packages/react/package.json#L46

So, if you have 2.5.5 then: 
<img width="1293" alt="image" src="https://github.com/user-attachments/assets/28ccdb78-f826-4c07-bc55-9a815f2da21c">
and if you have 2.5.4 then: 
<img width="1315" alt="image" src="https://github.com/user-attachments/assets/48fa9484-8509-4c49-ae56-e0066e01b13e">

So if you install anything with 2.5.5 (i.e. you deleted a package-lock & pulled in a latest version) then it will pretty much force everything to have to be 2.5.5 (because we only support these versions together not mixed & matched)

So the problem is that you deleted your package-lock, which would hold the versions back to the ones you specified. This is how NPM package resolution works, this is a valid packaging strategy used by other projects as well.

**Comment by Heet-Bhalodiya** [comment]
Hi @nperez0111,

Thanks for your quick response.

We build templates and use this editor on multiple pages and apps. Now, we have to fix package versions in our `package.json` file in order to avoid any unexpected package updates and hence, avoiding any template installation issues. We also provide `pnpm-lock.yaml` file inside the package but there are multiple cases reported by our customers where they try any other installer like `NPM` or `yarn` where either installation stop working or they get to see these warnings for `pnpm`

Here is an example of how we provide the `package.json` file with fixed versions:
```json
"dependencies": {
  "@tiptap/extension-color": "2.5.4",
  "@tiptap/extension-list-item": "2.5.4",
  "@tiptap/extension-placeholder": "2.5.4",
  "@tiptap/extension-text-align": "2.5.4",
  "@tiptap/extension-text-style": "2.5.4",
  "@tiptap/extension-underline": "2.5.4",
  "@tiptap/pm": "2.5.4",
  "@tiptap/react": "2.5.4",
  "@tiptap/starter-kit": "2.5.4",	
}
```

Since we do not provide a lock file for other package managers, using `npm` result in the following installation errors:

![image](https://github.com/user-attachments/assets/ae131a79-1552-4cfd-81d5-395694ce1999)

Is there any way for using fix version like `2.5.4`(or any other version except latest) and avoid the above errors?

We are ready to update to the latest version of the `tiptap` and provide to our customers. Until we have not provided the latest package to our customers, they will be facing the issue shown above in the screen shot. Do you have any solution for this?

Alternatively, we kindly request that you should consider less frequent updates for smoother product experience, preventing such warnings from occurring with each minor release.

Hope you understand my concerns.

**Comment by nperez0111** [comment]
We are not going to consider changing the versioning scheme as it reflects what we want as maintainers. The packages could be incompatible & it is more work to deal with all the different possible combinations of tiptap versions.

What you could do to fix your problem is instead of relying on peer dependency resolution you could explicitly declare that your project needs the packages that are currently be downloading as peer dependencies.

For example, `@tiptap/core@2.5.5` is being downloaded because `@tiptap/extension-color@2.5.4` declares `@tiptap/core` as a peer dep and so it is being resolved to the newer version because of the rules I talked about earlier.
